src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -278,48 +309,25 @@ def list(self, sources=None):
                      'hostname': (str, 'Hostname'),
                      'addr': (str, 'Network Address'),
                      'labels': ([str], 'Host Labels'),
-                     'status': (str, 'Status of the Host')
+                     'maintenance': (bool, 'Enable Host Maintenance')
                  },
                  responses={200: None, 204: None})
     @RESTController.MethodMap(version='0.1')
     def create(self, hostname: str,
                addr: Optional[str] = None,
                labels: Optional[List[str]] = None,
-               status: Optional[str] = None):  # pragma: no cover - requires realtime env
-        orch_client = OrchClient.instance()
-        self._check_orchestrator_host_op(orch_client, hostname, True)
-        orch_client.hosts.add(hostname, addr, labels, status)
-    create._cp_config = {'tools.json_in.force': False}  # pylint: disable=W0212
+               maintenance: bool = False):  # pragma: no cover - requires realtime env
```

Comment:
I see the main issue here is if `status==maintenance` then it should instead ` orch_client.hosts.enter_maintenance()` as you did in `create_host()` . So why replace the variable? How about instead having a conditional statement like 
```
if status == 'maintenance':
            orch_client.hosts.enter_maintenance(hostname, False)
        else:
            orch_client.hosts.add(hostname, addr, labels, status)

Reply:
I updated the logic as per your suggestion. But the else part you mentioned is not the correct approach. A host can only be put to maintenance once it is added to the cluster.